router.go

```diff
@@ -623,18 +625,44 @@ func (r *Router) register(method, path string, op *openAPIOperation) {
 					}
 				}
 
-				if strings.HasPrefix(r.ContentType, "application/json") || strings.HasSuffix(r.ContentType, "+json") {
+				ct := r.ContentType
+
+				// Allow content-negotiation to override the default for known structured
+				// responses.
+				if strings.Contains(ct, "json") || strings.Contains(ct, "yaml") || strings.Contains(ct, "cbor") {
+					if accept := c.GetHeader("Accept"); accept != "" {
+						best := selectQValue(accept, []string{"application/cbor", "application/json", "application/yaml", "application/x-yaml"})
```

Comment:
Readability suggestion: Make the list of accepted values a constant so it can be easily extracted and returned to a client querying about capabilities. Also, this helps minimize the spread of magic values.